mempool/src/mempool.rs

```diff
@@ -162,14 +207,48 @@ impl Mempool {
         *executor_handle = Some(abort_handle);
     }
 
-    /// Stops the mempool executor
+    /// Starts the control mempool executor with a custom transaction stream
+    ///
+    /// Once this function is called, the mempool executor is spawned.
+    /// The executor won't subscribe to the transaction topic from the network but will use the provided transaction
+    /// stream instead.
+    pub async fn start_control_executor_with_txn_stream<N: Network>(
+        &self,
+        txn_stream: BoxStream<'static, (Transaction, <N as Network>::PubsubId)>,
+        network: Arc<N>,
+    ) {
+        let mut executor_handle = self.control_executor_handle.lock().await;
```

Comment:
Yes I thought about that, but since the control/regular executor are really distinct entities, then even if we only have a single function that receives a parameter to decide which executor to create, we would still need to instantiate both executors, both handles, etc. so there are not much gains I think